--- /usr/share/arduino/hardware/arduino/avr/libraries/SoftwareSerial/src/SoftwareSerial.cpp	2016-05-10 05:18:16.000000000 -0400
+++ src/SoftwareSerial.cpp	2016-05-23 21:30:11.627766943 -0400
@@ -224,6 +224,7 @@
   }
 }
 
+#if 0
 #if defined(PCINT0_vect)
 ISR(PCINT0_vect)
 {
@@ -242,6 +243,12 @@
 #if defined(PCINT3_vect)
 ISR(PCINT3_vect, ISR_ALIASOF(PCINT0_vect));
 #endif
+#endif
+
+ISR(PCINT2_vect)
+{
+  SoftwareSerial::handle_interrupt();
+}
 
 //
 // Constructor
@@ -272,8 +279,8 @@
   // the pin would be output low for a short while before switching to
   // output high. Now, it is input with pullup for a short while, which
   // is fine. With inverse logic, either order is fine.
-  digitalWrite(tx, _inverse_logic ? LOW : HIGH);
-  pinMode(tx, OUTPUT);
+  //digitalWrite(tx, _inverse_logic ? LOW : HIGH);
+  //pinMode(tx, OUTPUT);
   _transmitBitMask = digitalPinToBitMask(tx);
   uint8_t port = digitalPinToPort(tx);
   _transmitPortRegister = portOutputRegister(port);
